### ISSUE TITLE
How can I train the model efficiently?

### ISSUE THREAD

**Issue by RitvikN** [microsoft/AirSim-Drone-Racing-VAE-Imitation#4]
Hi,

I've tried to train the model using train_cmvae.py and the 50k dataset on both my personal linux laptop as well as a dedicated DL machine. However, in both cases the training only occurs on my CPU and does not make use of the GPU. 

As a result, the process is killed after some time after exceeding available memory on my laptop. The training runs very slowly on the DL machine, completing 1 epoch after several minutes. Since the base parameters are BATCH_SIZE=32 and EPOCHS=10000, at this rate it would take over half a year to finish training.

Could you please mention how you were able to train the model or how I can train it faster?

Thank you.

**Comment by rogeriobonatti** [comment]
Hi, the training scripts should automatically start training on your GPU if you installed tensorflow correctly. Are you sure you installed tensorflow with the GPU version? The regular one will only allow training on CPU. You may also need to check if you have enough RAM to hold the entire dataset during training, but if you started training, this is probably fine.

Also, I set epochs to 10000 as an arbitrary number. It actually only needs about 30-35 epochs to converge to a good representation. I acknowledge this is confusing, and I'll update the number on the repo.

**Comment by RitvikN** [comment]
> Hi, the training scripts should automatically start training on your GPU if you installed tensorflow correctly. Are you sure you installed tensorflow with the GPU version? The regular one will only allow training on CPU. You may also need to check if you have enough RAM to hold the entire dataset during training, but if you started training, this is probably fine.

I was able to get it working on a different machine, thanks.

> Also, I set epochs to 10000 as an arbitrary number. It actually only needs about 30-35 epochs to converge to a good representation. I acknowledge this is confusing, and I'll update the number on the repo.

Thank you for clarifying. The loss was already quite low after training for 100 epochs.

Another question, is there any way to modify the vae_env airsim binaries to change the orientation of the gates? I would like to see the performance on different kinds of tracks and not just a circular one. I know gate displacement is possible when using the recording environment but that too is still a circular track.

Lastly, is there any way to run the environments/project on Windows?

**Comment by rogeriobonatti** [comment]
I've never tried running on Windows, just on Ubuntu.
Not sure if the Linux binaries can be executed from Windows... the Python code shouldn't be an issue.
